full-stack-tests/core/src/frontend/hub/SheetViewState.test.ts

```diff
@@ -169,3 +174,94 @@ describe("Sheet views (#integration)", () => {
+  });
+
+  describe.only("ViewAttachments", () => {
+    function createBlankConnection(name = "test-blank-connection",
+      location = Cartographic.fromDegrees({ longitude: -75.686694, latitude: 40.065757, height: 0 }),
+      extents = new Range3d(-1000, -1000, -100, 1000, 1000, 100),
+      iTwinId = Guid.createValue()): BlankConnection {
+      return BlankConnection.create({ name, location, extents, iTwinId });
+    }
+
+    interface BlankViewportOptions {
+      /** Height in pixels. Default 100. */
+      height?: number;
+      /** Width in pixels. Default 100. */
+      width?: number;
+      /** iModel. If undefined, a new blank connection will be created. */
+      iModel?: BlankConnection;
+      /** The position of the containing div. */
+      position?: "absolute";
+    }
+
+    /** Open a viewport for a blank spatial view.
+     * @internal
+     */
+    function openBlankViewport(options?: BlankViewportOptions): ScreenViewport {
+      const height = options?.height ?? 100;
+      const width = options?.width ?? 100;
+      const iModel = options?.iModel ?? createBlankConnection();
+
+      const parentDiv = document.createElement("div");
+      const hPx = `${height}px`;
+      const wPx = `${width}px`;
+
+      parentDiv.setAttribute("height", hPx);
+      parentDiv.setAttribute("width", wPx);
+      parentDiv.style.height = hPx;
+      parentDiv.style.width = wPx;
+
+      if (options?.position)
+        parentDiv.style.position = options.position;
+
+      document.body.appendChild(parentDiv);
+
+      const view = SpatialViewState.createBlank(iModel, { x: 0, y: 0, z: 0 }, { x: 1, y: 1, z: 1 });
+
+      class BlankViewport extends ScreenViewport {
+        public ownedIModel?: BlankConnection;
+
+        public override[Symbol.dispose](): void {
+          if (!this.isDisposed) {
+            document.body.removeChild(this.parentDiv);
+            super[Symbol.dispose]();
+            this.ownedIModel?.closeSync();
+          }
+        }
+      }
+
+      const viewport = BlankViewport.create(parentDiv, view) as BlankViewport;
+      if (undefined === options?.iModel)
+        viewport.ownedIModel = iModel;
+
+      return viewport;
+    }
+
+    it("areAllTileTreesLoaded should return true when attachments are outside of the viewed extents", async () => {
+      if (!process.env.IMODELJS_CORE_DIRNAME)
+        throw new Error("IMODELJS_CORE_DIRNAME not set");
+
+      const vp = openBlankViewport({ height: 100, width: 100 });
+      const sheetViewStateProps = await coreFullStackTestIpc.insertViewAttachmentAndGetSheetViewProps();
+      expect(sheetViewStateProps.sheetAttachments).not.to.be.undefined;
+      expect(sheetViewStateProps.sheetAttachments!.length).to.equal(1);
+      const newSheetView = SheetViewState.createFromProps(sheetViewStateProps, await BriefcaseConnection.openStandalone(path.join(process.env.IMODELJS_CORE_DIRNAME, "core/backend/lib/cjs/test/assets/sheetViewTest.bim")));
```

Comment:
Your viewport is viewing a `BlankConnection`. Your `SheetViewState` is from a `BriefcaseConnection`. They have to be the same iModel. Let me try to fix it.

Reply:
Hacked that in.